docs/template_plugin/tests/functional/op_reference/comparison.cpp

```diff
@@ -0,0 +1,149 @@
+                     const ngraph::element::Type& iType, const ngraph::element::Type& oType, const std::vector<IT>& iValues1, const std::vector<IT>& iValues2,
+                     const std::vector<OT>& oValues)
+        : comparisonType(comp),
+          pshape1(input_shape1),
+          pshape2(input_shape2),
+          inType(iType),
+          outType(oType),
+          inputData1(CreateBlob(iType, iValues1)),
+          inputData2(CreateBlob(iType, iValues2)),
+          refData(CreateBlob(oType, oValues)) {}
+    ComparisonTypes comparisonType;
+    ngraph::PartialShape pshape1;
+    ngraph::PartialShape pshape2;
+    ngraph::element::Type inType;
+    ngraph::element::Type outType;
+    InferenceEngine::Blob::Ptr inputData1;
+    InferenceEngine::Blob::Ptr inputData2;
+    InferenceEngine::Blob::Ptr refData;
+};
+
+class ReferenceComparisonLayerTest : public testing::TestWithParam<RefComparisonParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params.comparisonType, params.pshape1, params.pshape2, params.inType, params.outType);
+        inputData = {params.inputData1, params.inputData2};
+        refOutData = {params.refData};
+    }
+    static std::string getTestCaseName(const testing::TestParamInfo<RefComparisonParams>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "comparisonType=" << param.comparisonType << "_";
+        result << "inpt_shape1=" << param.pshape1 << "_";
+        result << "inpt_shape2=" << param.pshape2 << "_";
+        result << "iType=" << param.inType << "_";
+        result << "oType=" << param.outType;
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(ComparisonTypes& comp_op_type, PartialShape& input_shape1, const PartialShape& input_shape2,
+                                                    const element::Type& input_type, const element::Type& expected_output_type) {
+        const auto in = std::make_shared<op::Parameter>(input_type, input_shape1);
+        const auto in2 = std::make_shared<op::Parameter>(input_type, input_shape2);
+        const auto comp = ngraph::builder::makeComparison(in, in2, comp_op_type);
+        return std::make_shared<Function>(NodeVector {comp}, ParameterVector {in, in2});
+    }
+};
+
+TEST_P(ReferenceComparisonLayerTest, CompareWithHardcodedRefs) {
+    Exec();
+}
+
+template <element::Type_t IN_ET>
+std::vector<RefComparisonParams> generateComparisonParams(const ngraph::element::Type& type) {
+    using T = typename element_type_traits<IN_ET>::value_type;
+    std::vector<RefComparisonParams> compParams {
+        // 1D // 2D // 3D // 4D
+        RefComparisonParams(ComparisonTypes::LESS, ngraph::PartialShape {2, 2}, ngraph::PartialShape {2, 2}, type, ngraph::element::boolean,
```

Comment:
I propose to have separate file of each operation.

Reply:
Good idea, I left the `generateComparisonCombinedParams` function duplicated for each op, as it seems like a good way to constrain the tested types for a single op, and not the whole Comparison group.